### PR TITLE
Desk Bell

### PR DESCRIPTION
**Pull request by Birdtalon**
# About the pull request

Using my maintainer privilege to add this small feature during the freeze. A desk bell.

Wrench to secure it and use it.
Bell can break at a 1% chance for every 100 rings. In which case a screwdriver will fix it.

dreamseeker_2023-12-29_00-39-47
dreamseeker_2023-12-29_00-39-56

Credit:
Modified code and desk_bell.ogg from [Paradise SS13](https://github.com/ParadiseSS13/Paradise)
Sprite from Kat 'APC' Smith - darklordcabbage on discord.
Animated sprite from Thwomp

# Explain why it's good for the game
# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl: Birdtalon, Kat 'APC' Smith, Thwomp
add: Desk bell for cargo and medbay.
soundadd: Desk bell sound effect.
/:cl:

### CODE REVIEW

File: code/modules/paperwork/desk_bell.dm [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "desk_bell"
+	w_class = SIZE_SMALL
+	embeddable = FALSE
+	/// The amount of times this bell has been rang, used to check the chance it breaks.
+	var/times_rang = 0
+	/// Is this bell broken?
+	var/broken_ringer = FALSE
+	/// Holds the time that the bell can next be rang.
+	var/ring_cooldown = 0
```
Comment (Cthulhu80) [inline code/modules/paperwork/desk_bell.dm:15]: COOLDOWN_DECLARE(ring_cooldown)

File: code/modules/paperwork/desk_bell.dm [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+	/// The sound the bell makes.
+	var/ring_sound = 'sound/misc/desk_bell.ogg'
+
+/obj/item/desk_bell/attack_hand(mob/living/user)
+	if(!ishuman(user))
+		return FALSE
+	if(!anchored)
+		return ..()
+	if(ring_cooldown > world.time)
+		return FALSE
```
Comment (Cthulhu80) [inline code/modules/paperwork/desk_bell.dm:27]: COOLDOWN_FINISHED(ring_cooldown)

File: code/modules/paperwork/desk_bell.dm [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+	if(!ring_bell(user))
+		to_chat(user, SPAN_NOTICE("[src] is silent. Some idiot broke it."))
+		return FALSE
+	ring_cooldown = world.time + ring_cooldown_length
```
Comment (Cthulhu80) [inline code/modules/paperwork/desk_bell.dm:31]: COOLDOWN_START(src, ring_cooldown, ring_cooldown_length)

**Comment by JackieEstegado** [comment]
Please for the love of all that is hearing, increase the cooldown to AT LEAST 3 seconds.

File: maps/map_files/USS_Almayer/USS_Almayer.dmm
Comment (Nanu308) [inline maps/map_files/USS_Almayer/USS_Almayer.dmm:1]: mapping lgtm

**Review by Nanu308 — APPROVED** [review]

**Comment by Birdtalon** [comment]
Have made further map changes @Nanu308 added a couple more bells to various areas at request of the sprite artist.

**Review by nauticall — CHANGES_REQUESTED** [review]
It's cute, but the bell is _way_ too high off the stand and the stand is also way too big. You only really need a pixel of locomotion for the animation to be visible from a distance, and can achieve that even with a small stand. See below for my recommendation.
image


image

Also note: avoid using diagonals that zig-zag like this. Try using an intermediate color to break up the line instead, or avoid zig-zags entirely and use straight diagonals.

**Comment by Birdtalon** [comment]
@nauticall thanks for the review, I will pass this onto the spriter and get them to adjust the sprite accordingly.

File: maps/map_files/USS_Almayer/USS_Almayer.dmm
Comment (Nanu308) [inline maps/map_files/USS_Almayer/USS_Almayer.dmm:1]: lgtm

**Review by Nanu308 — APPROVED** [review]

**Pull request closed without merging**